azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_stages_provisioning_mqtt.py

```diff
@@ -31,7 +31,11 @@ def __init__(self):
 
     def _run_op(self, op):
 
-        if isinstance(op, pipeline_ops_provisioning.SetSymmetricKeySecurityClientArgs):
+        if isinstance(
+            op, pipeline_ops_provisioning.SetSymmetricKeySecurityClientArgs
+        ) or isinstance(
+            op, pipeline_ops_provisioning.SetSecurityClientArgs
+        ):  # TODO Generic Args , then only 1 case ?
```

Comment:
> ):  # TODO Generic Args , then only 1 case ? [](start = 7, length = 45)

If I understand this TODO, then I think I agree.  It looks like SetSymmetricKeySecurityClientArgs and SetSecurityClientArgs could/should be merged and made into one operation.

Reply:
+1, and of course I see this after spending 20 minutes reading through the code and coming to this conclusion myself smh

---
In reply to: [291684999](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291684999) [](ancestors = 291684999)